src/transform.rs

```diff
@@ -0,0 +1,161 @@
+        Err(ZkError::BadVersion) => Ok(Err(error::Delete::BadVersion { expected: version })),
+        Err(e) => Err(format_err!("delete call failed: {:?}", e)),
+    }
+}
+
+pub fn get_acl(
+    res: Result<Response, ZkError>,
+) -> Result<Result<(Vec<Acl>, Stat), error::GetAcl>, failure::Error> {
+    match res {
+        Ok(Response::GetAcl { acl, stat }) => Ok(Ok((acl, stat))),
+        Ok(r) => bail!("got non-acl response to a get_acl request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(Err(error::GetAcl::NoNode)),
+        Err(e) => Err(format_err!("get_acl call failed: {:?}", e)),
+    }
+}
+
+pub fn set_acl(
+    version: i32,
+    res: Result<Response, ZkError>,
+) -> Result<Result<Stat, error::SetAcl>, failure::Error> {
+    match res {
+        Ok(Response::Stat(stat)) => Ok(Ok(stat)),
+        Ok(r) => bail!("got non-stat response to a set_acl request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(Err(error::SetAcl::NoNode)),
+        Err(ZkError::BadVersion) => Ok(Err(error::SetAcl::BadVersion { expected: version })),
+        Err(ZkError::InvalidACL) => Ok(Err(error::SetAcl::InvalidAcl)),
+        Err(ZkError::NoAuth) => Ok(Err(error::SetAcl::NoAuth)),
+        Err(e) => Err(format_err!("set_acl call failed: {:?}", e)),
+    }
+}
+
+pub fn exists(res: Result<Response, ZkError>) -> Result<Option<Stat>, failure::Error> {
+    match res {
+        Ok(Response::Stat(stat)) => Ok(Some(stat)),
+        Ok(r) => bail!("got a non-create response to a create request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(None),
+        Err(e) => bail!("exists call failed: {:?}", e),
+    }
+}
+
+pub fn get_children(res: Result<Response, ZkError>) -> Result<Option<Vec<String>>, failure::Error> {
+    match res {
+        Ok(Response::Strings(children)) => Ok(Some(children)),
+        Ok(r) => bail!("got non-strings response to get-children: {:?}", r),
+        Err(ZkError::NoNode) => Ok(None),
+        Err(e) => Err(format_err!("get-children call failed: {:?}", e)),
+    }
+}
+
+pub fn get_data(res: Result<Response, ZkError>) -> Result<Option<(Vec<u8>, Stat)>, failure::Error> {
+    match res {
+        Ok(Response::GetData { bytes, stat }) => Ok(Some((bytes, stat))),
+        Ok(r) => bail!("got non-data response to get-data: {:?}", r),
+        Err(ZkError::NoNode) => Ok(None),
+        Err(e) => Err(format_err!("get-data call failed: {:?}", e)),
+    }
+}
+
+pub fn check(
+    version: i32,
+    res: Result<Response, ZkError>,
+) -> Result<Result<(), error::Check>, failure::Error> {
+    match res {
+        Ok(Response::Empty) => Ok(Ok(())),
+        Ok(r) => bail!("got a non-check response to a check request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(Err(error::Check::NoNode)),
+        Err(ZkError::BadVersion) => Ok(Err(error::Check::BadVersion { expected: version })),
+        Err(e) => bail!("check call failed: {:?}", e),
+    }
+}
+
+#[derive(Debug)]
+pub enum RequestLite {
```

Comment:
I assume this can just be `pub(crate)`?
Also, I'd like for us to give it a more descriptive name. If I understand correctly, it's there so that you can remember what requests you sent out in a multi? Maybe `RequestMarker`?

Reply:
> I assume this can just be pub(crate)?

Yep, done. Sorry, still getting used to the new `pub(MODIFIER)` visibility specifiers.

> Also, I'd like for us to give it a more descriptive name. If I understand correctly, it's there so that you can remember what requests you sent out in a multi? Maybe RequestMarker?

Yep, exactly. It saves you from needing to clone the whole request when all you needed to remember was the version. `RequestMarker` sounds good to me.

Reply:
Hehe, no worries! I haven't spotted any other `pub`s, but we should be careful that we don't accidentally add any. Luckily it should be as easy as searching for `pub ` on `master` and on your branch, and comparing the two!